### PR TITLE
Added configurations for VSCode users

### PR DESCRIPTION
**Pull request by geard-dev**
# Added configurations for VSCode users

The purpose of this was to make developing easier for VSCode users by using VSCode Tasks to let you use the different commands (for building and installing devDependencies) using a one click method inside VSCode

## Commit 1 (f3abd39)

- Created .vscode\tasks.json

- Edited .gitignore to not ignore .vscode/tasks.json, but ignore all other files in .vscode/

## Commit 2 (d91032a)

- Correction: Changed .vscode/tasks.json to .vscode/launch.json in the filename and .gitignore

## Commit 3 (e384d14)

- Wrote the different configurations in .vscode/launch.json

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+			"command": "npm run debug"
+		},
+		{
+			"name": "Build for Production",
+			"type": "node-terminal",
+			"request": "launch",
+			"command": "npm run production"
+		},
+		{
+			"name": "Install devDependencies",
```
Comment (Gusted) [inline .vscode/launch.json:16]: I think `Install NPM Dependencies` sounds better.

**Review by Gusted — CHANGES_REQUESTED** [review]
Great addition!

File: .vscode/launch.json (hunk 1 above)
Comment (geard-dev) [reply .vscode/launch.json:16]: Alright, Ill make a quick commit to change it

**Comment by geard-dev** [comment]
Should I also update the README?

**Comment by Gusted** [comment]
> Should I also update the README?

Sure things.

**Review by Gusted — APPROVED** [review]

**Comment by geard-dev** [comment]
I think there might be a problem with the build scripts themselves, I tried testing building it using both the VSCode UI and the commands and I got the same error. Everything is up to date; my NPM and the packages

**Comment by Gusted** [comment]
Could you show the error!

**Comment by geard-dev** [comment]
Running npm run production:
```Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\Geard\Documents\GitHub\Deadname-Remover> npm run production

> deadname-remover@1.1.2 production C:\Users\Geard\Documents\GitHub\Deadname-Remover
> node tasks/build.js --production

events.js:287
      throw er; // Unhandled 'error' event
      ^

Error: spawn C:\Users\Geard\Documents\GitHub\Deadname-Remover\node_modules\esbuild\esbuild.exe ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn C:\\Users\\Geard\\Documents\\GitHub\\Deadname-Remover\\node_modules\\esbuild\\esbuild.exe',
  path: 'C:\\Users\\Geard\\Documents\\GitHub\\Deadname-Remover\\node_modules\\esbuild\\esbuild.exe',
  spawnargs: [ '--service=0.7.11' ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! deadname-remover@1.1.2 production: `node tasks/build.js --production`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the deadname-remover@1.1.2 production script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Geard\AppData\Roaming\npm-cache\_logs\2020-10-09T00_15_56_239Z-debug.log```

Logfile:
```log
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\Geard\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'production'
1 verbose cli ]
2 info using npm@6.14.8
3 info using node@v12.16.3
4 verbose run-script [ 'preproduction', 'production', 'postproduction' ]
5 info lifecycle deadname-remover@1.1.2~preproduction: deadname-remover@1.1.2
6 info lifecycle deadname-remover@1.1.2~production: deadname-remover@1.1.2
7 verbose lifecycle deadname-remover@1.1.2~production: unsafe-perm in lifecycle true
8 verbose lifecycle deadname-remover@1.1.2~production: PATH: C:\Users\Geard\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\Geard\Documents\GitHub\Deadname-Remover\node_modules\.bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\nodejs\;C:\Program Files\WorldPainter;C:\Program Files\dotnet\;C:\Program Files\Java\jdk1.8.0_251\bin;C:\Program Files\Git\cmd;C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin;C:\Users\Geard\.cargo\bin;C:\Users\Geard\AppData\Local\Programs\Python\Python38\Scripts\;C:\Users\Geard\AppData\Local\Programs\Python\Python38\;C:\Users\Geard\AppData\Local\Microsoft\WindowsApps;C:\Users\Geard\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Geard\AppData\Roaming\npm;C:\Users\Geard\AppData\Local\GitHubDesktop\bin;C:\Users\Geard\AppData\Local\Microsoft\WindowsApps;C:\Users\Geard\.dotnet\tools;C:\Program Files (x86)\GitHub CLI\
9 verbose lifecycle deadname-remover@1.1.2~production: CWD: C:\Users\Geard\Documents\GitHub\Deadname-Remover
10 silly lifecycle deadname-remover@1.1.2~production: Args: [ '/d /s /c', 'node tasks/build.js --production' ]
11 silly lifecycle deadname-remover@1.1.2~production: Returned: code: 1  signal: null
12 info lifecycle deadname-remover@1.1.2~production: Failed to exec production script
13 verbose stack Error: deadname-remover@1.1.2 production: `node tasks/build.js --production`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Users\Geard\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:310:20)
13 verbose stack     at ChildProcess.<anonymous> (C:\Users\Geard\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:310:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid deadname-remover@1.1.2
15 verbose cwd C:\Users\Geard\Documents\GitHub\Deadname-Remover
16 verbose Windows_NT 10.0.19041
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Geard\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "production"
18 verbose node v12.16.3
19 verbose npm  v6.14.8
20 error code ELIFECYCLE
21 error errno 1
22 error deadname-remover@1.1.2 production: `node tasks/build.js --production`
22 error Exit status 1
23 error Failed at the deadname-remover@1.1.2 production script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

Running production build using VSCode UI:
```
> deadname-remover@1.1.2 production C:\Users\Geard\Documents\GitHub\Deadname-Remover
> node tasks/build.js --production

c:\Users\Geard\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:255
Uncaught Error: spawn C:\Users\Geard\Documents\GitHub\Deadname-Remover\node_modules\esbuild\esbuild.exe ENOENT
<node_internals>/internal/errors.js:512
No debugger available, can not send 'variables'

c:\Users\Geard\AppData\Roaming\npm\node_modules\npm\lib\utils\error-handler.js:71
```

**Comment by Gusted** [comment]
Oh.. Your attaching a node debugger it seems like.. But that doesn't work with the build scripts and it shouldn't be any debugger attached to.

**Comment by geard-dev** [comment]
> Oh.. Your attaching a node debugger it seems like.. But that doesn't work with the build scripts and it shouldn't be any debugger attached to.

I'm not _that_ familiar with node so I don't really know what I'm doing, sorry about that

**Comment by Gusted** [comment]
Hmm, it seems like `launch.json` is specified made for Debugging?
Are you trying to add a `shortcut` to run commands, as for that VSCode is already native doing that

**Comment by geard-dev** [comment]
Oh, I've never really payed attention to the NPM scripts thing, yea this was supposed to be like a shortcut

**Pull request closed without merging**